practice-two/src/contexts/FormContext.jsx

```diff
@@ -0,0 +1,110 @@
+import { createContext, useState, useCallback, useContext } from 'react';
+import { Validation } from '@helpers';
+import { createCustomerService } from '@services';
+import { CustomerContext } from './CustomerContext';
+
+export const FormContext = createContext();
+
+export const FormProvider = ({ children }) => {
+  const [isShowForm, setIsShowForm] = useState(false);
+  //   const [isCreateFailed, setIsCreateFailed] = useState(false);
+  const [createResult, setCreateResult] = useState(null);
+  const { handleCreateCustomer } = useContext(CustomerContext);
+
+  const handleShowForm = useCallback(() => {
+    setIsShowForm(!isShowForm);
+  }, [isShowForm]);
+
+  const [formData, setFormData] = useState({
+    name: '',
+    avatar: '',
+    mail: '',
+    phoneNumber: '',
+    description: '',
+    address: '',
+    gender: 'Male',
+  });
+  const [errors, setErrors] = useState({
+    name: '',
+    avatar: '',
+    mail: '',
+    phoneNumber: '',
+    description: '',
+    address: '',
+  });
+
+  // Set value for form data
+  const handleChange = useCallback(
+    (e) => {
+      const { name, value } = e.target;
+      setFormData((prevFormData) => ({
+        ...prevFormData,
+        [name]: value,
+      }));
+    },
+    [setFormData]
+  );
+
+  // Check validate input when click outside
+  const handleBlur = useCallback(
+    (e) => {
+      const { name, value } = e.target;
+      const errorMessage = Validation(name, value);
+      setErrors((prevErrors) => ({
+        ...prevErrors,
+        [name]: errorMessage,
+      }));
+    },
+    [setErrors]
+  );
+
+  // Submit form
+  const handleSubmit = useCallback(
+    async (e) => {
+      e.preventDefault();
+
+      // Check validation form when submiting
+      const newErrors = {};
+      for (const field in formData) {
+        const errorMessage = Validation(field, formData[field]);
+        if (errorMessage) {
+          newErrors[field] = errorMessage;
+        }
+      }
+      setErrors(newErrors);
+
+      if (Object.keys(newErrors).length === 0) {
+        const response = await createCustomerService(formData);
+
+        if (response.error) {
+          handleShowForm();
+          setCreateResult('failed');
+          return;
+        } else {
+          handleShowForm();
+          handleCreateCustomer(response.data);
+          setCreateResult('success');
+        }
+      }
+    },
+    [formData, setErrors, handleShowForm]
+  );
+
+  const formContextValue = {
+    formData,
+    errors,
+    setErrors,
+    isShowForm,
+    handleShowForm,
+    handleChange,
+    handleBlur,
+    handleSubmit,
+    createResult,
+  };
+
+  return (
+    <FormContext.Provider value={formContextValue}>
```

Comment:
@datdragon777 Please clear Context, I don't think you need add FormContext in here
https://react.dev/reference/react/createContext